pcdsdevices/sqr1.py

```diff
@@ -0,0 +1,43 @@
+"""
+Ophyd class for square-one tri-sphere motion system (https://www.sqr-1.com/tsp.html)
+"""
+
+import logging
+
+from ophyd.device import Device
+from ophyd.device import FormattedComponent as FCpt
+from ophyd.pv_positioner import PVPositionerIsClose
+from ophyd.signal import EpicsSignal, EpicsSignalRO
+
+logger = logging.getLogger(__name__)
+
+
+class SQR1Axis(PVPositionerIsClose):
+    setpoint = FCpt(EpicsSignal, '{self.prefix}:TARGET:{self._axis}', kind='normal')
+    readback = FCpt(EpicsSignal, '{self.prefix}:TARGET:{self._axis}:RBV', kind='normal')
+    actuate = FCpt(EpicsSignal, '{self.prefix}:MOV', kind='normal')
+    actuate_value = 1
+    stop_signal = FCpt(EpicsSignal, '{self.prefix}:KILL', kind='normal')
+    stop_value = 1
+
+    llm = FCpt(EpicsSignalRO, '{self.prefix}:TARGET:{self._axis}:LLM', kind='normal')
+    hlm = FCpt(EpicsSignalRO, '{self.prefix}:TARGET:{self._axis}:HLM', kind='normal')
+
+    def __init__(self, axis, **kwargs):
+        self._axis = axis  # axis values {X,Y,Z,rX, rY, rZ}
+        super().__init__(**kwargs)
+
+        self.setpoint.put(self.readback.get())  # sync position setpoint with readback
```

Comment:
We should be really careful doing this, for three reasons:
1. We're blocking on two PV connections during `__init__`, and we're doing it 6 times for a maximum total of 12 possible timeouts.
2. We're causing a change to the control system on `__init__`. If a second user opens a session that contains a square 1 device while the first user is doing a scan, depending on the timing we can get very strange behavior.
3. This device will error out on `__init__` if run from anywhere outside of tmo, which may be disruptive.

What exactly is gained from doing this sync? Is there any way we can accomplish this during moves only instead of at initialization?

Reply:
This system has one move command PV for the six motors, therefore I initialize each motor setpoint value with its readback value to avoid moving one motor while other motor(s) setpoints are zeros for example, then they will move to unwanted positions.